### ISSUE TITLE
Allow renaming of unresolved locals

### ISSUE THREAD

**Issue by walterl** [clojure-lsp/clojure-lsp#1210]
**Is your feature request related to a problem?**
No, this is purely an enhancement.

**Describe the solution you'd like**
Say you end up with this function:

```clojure
(defn foo [i]
  (if (pos? x)
    (inc x)
    (dec x)))
```

clojure-lsp helpfully and correctly points out that all those `x`s are unresolved symbols.

It would be nice if, when renaming one of the `x`s (e.g. to `i`), clojure-lsp automatically rename all the other unresolved `x`s to `i`s too.

**Describe alternatives you've considered**
Workaround: rename `i` to `x`, and then use clojure-lsp to rename all `x`s to `i`s.

**Additional context**
Only unresolved vars in the local context should be renamed.

**Comment by ericdallo** [comment]
I took a quick look and one problem I found is that with current clj-kondo analysis we can know if that `x` is a local or a var defined elsewhere in the file, for example:
```clojure
x ;; not a local

(defn foo [i]
  (if (pos? x) ;; local
    (inc x)
    (dec x)))
```

All the 4 `x` there, clj-kondo return inside the `var-usages` bucket.
@borkdude do you think would work to make clj-kondo return the bucket of the undefined `x`s inside that `foo` function under `local-usages` bucket instead of `var-usages`?

**Comment by borkdude** [comment]
@ericdallo Isn't this already part of `:findings` of `:type :unresolved-symbol`? The only thing we should add to that finding is the symbol as data probably.

**Comment by ericdallo** [comment]
@borkdude it's available (besides the symbol), but I can't see how that would help here, what we need is a way to know all undefined local-usages to allow renaming, and not renaming all `x` of that file, so if we return a different bucket for that, probably would work for clojure-lsp

**Comment by borkdude** [comment]
@ericdallo Just because an unresolved symbol appears within an s-expression doesn't imply that it should have been a local. E.g.:

```
(def x 1)

(+ 1 2 xx) ;; typo, should have been x
```

I don't think this feature request can be reliably implemented and maybe isn't worth the investment: you can just use `replace-string` after all.

**Comment by ericdallo** [comment]
Yeah, I can see how that could leave some corner cases, and I think it's not worth the effort
